### PR TITLE
Adds name=all which allows a user to dump or import all data...

### PR DESCRIPTION
**Pull request by Jmainguy**
...bases at once, identical to mysqldump --all-databases

### CODE REVIEW

**Comment by Jmainguy** [comment]
This is a recreation of https://github.com/ansible/ansible/pull/8017

**Comment by Jmainguy** [comment]
Discussed on video call today, going to change from all_databases=true, to name=all

File: database/mysql/mysql_db.py [hunk 1]
```diff
@@ -271,6 +282,7 @@ def main():
             name=dict(required=True, aliases=['db']),
             encoding=dict(default=""),
             collation=dict(default=""),
+            all_databases=dict(default="no", choices=BOOLEANS, type='bool'),
```
Comment (abadger) [inline database/mysql/mysql_db.py]: Need to remove this now that we're using name=all instead.

File: database/mysql/mysql_db.py [hunk 2]
```diff
@@ -280,6 +292,7 @@ def main():
         module.fail_json(msg="the python mysqldb module is required")
 
     db = module.params["name"]
+    all_databases = False
```
Comment (abadger) [inline database/mysql/mysql_db.py]: I'd initialize this at the same point as we're setting it to True if db == all.

File: database/mysql/mysql_db.py [hunk 3]
```diff
@@ -60,7 +60,8 @@
     default: null
   state:
     description:
-      - The database state
+      - The database state 
+      - name=all May only be provided if I(state) is C(dump) or C(import). Works like --all-databases option for mysqldump (Added in 2.0)
```
Comment (abadger) [inline database/mysql/mysql_db.py]: I'd document this in the name parameter rather than in the state parameter.

**Comment by abadger** [comment]
Thanks for your work on this!

# Merged

Hi!

This has been merged in, and will also be included in the next major release.

If you or anyone else has any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!

**Pull request merged**